frontend/src/app/main/ui/shapes/custom_stroke.cljs

```diff
@@ -419,8 +419,13 @@
         stroke-props (-> (obj/new)
                          (obj/set! "id" (dm/fmt "strokes-%" (:id shape)))
                          (cond->
-                          (and (and (:blur shape) (-> shape :blur :hidden not)) (not (cph/frame-shape? shape)))
-                           (obj/set! "filter" (dm/fmt "url(#filter_blur_%)" render-id))))]
+                          ;; There is a blur
+                          (and (:blur shape) (-> shape :blur :hidden not) (not (cph/frame-shape? shape)))
+                          (obj/set! "filter" (dm/fmt "url(#filter_blur_%)" render-id))
+
+                          ;; There are any shadows and no fills
+                          (and (empty? (:fills shape)) (seq (->> (:shadow shape) (remove :hidden))) (not (cph/frame-shape? shape)))
```

Comment:
A little improvement, put the fastest predicates first on the and, for avoid doing complex stuff before is needed.

This applies to both branches of `cond->`